### PR TITLE
fix: Fix shapes clipping on Android

### LINKED ISSUE #13652: Samples app icons on Android are getting clipped
### Current behavior

image

### Expected behavior

_No response_

### How to reproduce it (as minimally and precisely as possible)

_No response_

### Workaround

_No response_

### Works on UWP/WinUI

None

### Environment

_No response_

### NuGet package version(s)

_No response_

### Affected platforms

_No response_

### IDE

_No response_

### IDE version

_No response_

### Relevant plugins

_No response_

### Anything else we need to know?

_No response_

### PR DESCRIPTION
**Pull request by Youssef1313**
Fixes #13652 

## PR Type

What kind of change does this PR introduce?

## What is the current behavior?

This code is causing incorrect clipping, for example:

image

Let's see if there will be any test failures.

### <samp>🤖 Generated by Copilot at a6002ca</samp>

Remove canvas clipping from `Shape.Android.cs` and delegate it to `FrameworkElement`. This improves the rendering of shapes with negative margins or offsets on Android.

## PR Checklist

Please check if your PR fulfills the following requirements:

- [ ] Docs have been added/updated which fit [documentation template](https://github.com/unoplatform/uno/blob/master/doc/.feature-template.md) (for bug fixes / features)
- [ ] [Unit Tests and/or UI Tests](https://github.com/unoplatform/uno/blob/master/doc/articles/uno-development/working-with-the-samples-apps.md) for the changes have been added (for bug fixes / features) (if applicable)
- [ ] Validated PR `Screenshots Compare Test Run` results.
- [ ] Contains **NO** breaking changes
- [ ] Associated with an issue (GitHub or internal) and uses the [automatic close keywords](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue).
- [ ] Commits must be following the [Conventional Commits](https://www.conventionalcommits.org/en/v1.0.0/#summary) specification.

## Other information

Internal Issue (If applicable):

### CODE REVIEW

**Comment by Youssef1313** [comment]
I reviewed screenshot compare results. The most meaningful change is:

**Before:**

Geometry_ClosedFigure_SamplesApp_Windows_UI_Xaml_Media_Geometry_ClosedFigurePage

**After:**

Geometry_ClosedFigure_SamplesApp_Windows_UI_Xaml_Media_Geometry_ClosedFigurePage

**Windows:**

image

**Review by dr1rrb — COMMENTED** [review]
LGTM

File: src/Uno.UI/UI/Xaml/Shapes/Shape.Android.cs [hunk 1]
```diff
@@ -34,19 +34,6 @@ protected override void OnDraw(Canvas canvas)
 				return;
 			}
 
-			//Drawing paths on the canvas does not respect the canvas' ClipBounds
```
Comment (jeromelaban) [inline src/Uno.UI/UI/Xaml/Shapes/Shape.Android.cs:37]: It would be best to include additional regression tests.

**Comment by MartinZikmund** [comment]
Very nice and blazing fast fix.

**Comment by Youssef1313** [comment]
Somehow, I'm not seeing the difference stated in https://github.com/unoplatform/uno/pull/13490#issuecomment-1704608537 locally.

**Comment by Youssef1313** [comment]
The only thing that is supposed to happen when pressed is that opacity is set to 0.7:

https://github.com/unoplatform/uno/blob/80ef443bf46ac0414e044775ae61a08d5d076f07/src/SamplesApp/SamplesApp.UnitTests.Shared/Controls/UITests/Views/Styles/Controls/RadioButton.xaml#L50-L55

Maybe it's related to #13394? The view that gets clipped is a *child* of the view that has the opacity. So maybe #13394 was only a partial fix, somehow? @carldebilly

File: src/Uno.UI/UI/Xaml/Shapes/Shape.Android.cs (hunk 1 above)
Comment (Youssef1313) [reply src/Uno.UI/UI/Xaml/Shapes/Shape.Android.cs:39]: @jeromelaban I wasn't able to isolate a minimal repro for testing.

Comment (dr1rrb) [reply src/Uno.UI/UI/Xaml/Shapes/Shape.Android.cs:39]: Well you could test the `ClosedFigure` sample you found in the screenshot diff https://github.com/unoplatform/uno/pull/13490#issuecomment-1704608537 and validate that you have black pixels in the "title bar" of the page.

**Review by dr1rrb — APPROVED** [review]

File: src/Uno.UI/UI/Xaml/Shapes/Shape.Android.cs (hunk 1 above)
Comment (Youssef1313) [reply src/Uno.UI/UI/Xaml/Shapes/Shape.Android.cs:39]: @dr1rrb Somehow, I'm not seeing the new behavior locally. So `ClosedFigure` is still broken :/

**Comment by dr1rrb** [comment]
> Somehow, I'm not seeing the difference stated in [#13490 (comment)](https://github.com/unoplatform/uno/pull/13490#issuecomment-1704608537) locally.
> 
> The clipping of "Recents" is fixed with this change, but when it's pressed, the clipping happens, and goes away when released. I have no idea what happens exactly on the press.
> 
> I also can't figure out a test yet.

Android touch event dispatch does not consider elements that are rendering themselves out of their bounds ... that's why the menu is being pressed when you click on it (it's the main reason why you worked on the android pointers dispatch ... we need to continue on this PR BTW). However I've no idea why the clipping is kicking-in on press.

**Comment by Youssef1313** [comment]
@dr1rrb The clipping happens on press because of opacity change. This is related to https://github.com/unoplatform/uno/pull/13394. But it seems that https://github.com/unoplatform/uno/pull/13394 wasn't enough of a fix. That is why I'm changing `hasOverlappingRendering` in this PR

File: src/Uno.UI/UI/Xaml/Shapes/Shape.Android.cs (hunk 1 above)
Comment (jeromelaban) [reply src/Uno.UI/UI/Xaml/Shapes/Shape.Android.cs:39]: @Youssef1313 is there something left to do here?

Comment (Youssef1313) [reply src/Uno.UI/UI/Xaml/Shapes/Shape.Android.cs:39]: Ideally, a test, but since I couldn't figure out one, I think it should be okay to merge as is.

**Comment by jeromelaban** [comment]
@Youssef1313 this test looks suspicious:

https://dev.azure.com/uno-platform/Uno%20Platform/_build/results?buildId=94299&view=ms.vss-test-web.build-test-results-tab&runId=2511100&resultId=100614&paneView=attachments

**Comment by Youssef1313** [comment]
Draft to review `UIElement_Layout_Constrains`

**Comment by Youssef1313** [comment]
Replaced by https://github.com/unoplatform/uno/pull/13931.

**Pull request closed without merging**